src/commands/functions/create.js

```diff
@@ -373,6 +367,42 @@ const createFunctionAddon = async function ({ api, addons, siteId, addonName, si
   }
 }
 
+const injectEnvVariables = async ({ context }) => {
+  const { log, warn, error, netlify } = context
+  const { api, site } = netlify
+  const { teamEnv, addonsEnv, siteEnv, dotFilesEnv } = await getSiteInformation({
+    api,
+    site,
+    warn,
+    error,
+  })
+  await addEnvVariables({ log, teamEnv, addonsEnv, siteEnv, dotFilesEnv })
+}
+
+const handleOnComplete = async ({ context, onComplete }) => {
+  if (onComplete) {
+    await injectEnvVariables({ context })
+    await onComplete.call(context)
+  }
+}
+
+const handleAddonDidInstall = async ({ addonCreated, addonDidInstall, context, fnPath }) => {
+  if (addonCreated && addonDidInstall) {
+    const { confirmPostInstall } = await inquirer.prompt([
+      {
+        type: 'confirm',
+        name: 'confirmPostInstall',
+        message: `This template has an optional setup script that runs after addon install. This can be helpful for first time users to try out templates. Run the script?`,
+        default: false,
+      },
+    ])
+    if (confirmPostInstall) {
+      await injectEnvVariables({ context })
```

Comment:
This was done so the `addonDidInstall` handler runs in the same context as `netlify dev`, but was missing dotenv envs.
This PR fixes that and also optimises to run `injectEnvVariables` only after user confirmation